slang-checks/src/main/java/com/sonarsource/slang/checks/BooleanLiteralCheck.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.BinaryExpressionTree;
+import com.sonarsource.slang.api.IfTree;
+import com.sonarsource.slang.api.LiteralTree;
+import com.sonarsource.slang.api.Tree;
+import com.sonarsource.slang.api.UnaryExpressionTree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.check.Rule;
+
+@Rule(key = "S1125")
+public class BooleanLiteralCheck implements SlangCheck {
+  private static final List<String> BOOLEAN_LITERALS = Arrays.asList("true", "false");
```

Comment:
I'm not sure that these values will be valid for all languages. Strictly speaking, we should define BooleanLiteralTree. Let's keep it like that for the moment...